src/orgwarden/repo_crawler.py

```diff
@@ -17,50 +18,81 @@ def __init__(self, hostname: str, message: str):
 
 
 def fetch_org_repos(
-    org_name: str,
-    hostname: str,
+    org_name: str, hostname: str, gh_pat: str | None = None
 ) -> list[Repository]:
     """
     Returns a `Repository` list containing the specified organization's public, non-forked repositories.
-    Raises an `APIError` if an error occurs while fetching repositories.
-    Raises a `FileNotFoundError` if the GitHub CLI is not installed.
+    Raises an `AuthError` if OrgWarden lacks authorization with GitHub API.
+    Raises an `APIError` if an other error occurs while fetching repositories, or the JSON response does not match expected schema.
     """
 
     if not org_name:
         raise ValueError("org_name is an empty string.")
     if not hostname:
         raise ValueError("hostname is an empty string.")
 
-    # check if gh cli is installed
-    if shutil.which("gh") is None:
-        raise FileNotFoundError("GitHub CLI is not installed.")
+    BASE_URL = (
+        "https://api.github.com"
+        if "github.com" in hostname or "www.github.com" in hostname
+        else f"https://{hostname}/api/v3"
+    )
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    if not gh_pat:
+        try:  # Load PAT to avoid rate-limits in local testing
+            import dotenv
+            import os
+
+            dotenv.load_dotenv()
+            gh_pat = os.environ.get("GITHUB_PAT")
+        except Exception:  # pragma: no cover
+            pass
+    if gh_pat:
+        headers["Authorization"] = f"Bearer {gh_pat}"
 
     org_repo_entries: list[dict] = []  # unfiltered api response
 
-    res = subprocess.run(
-        f"gh api orgs/{org_name}/repos --hostname {hostname} --paginate",
-        shell=True,
-        capture_output=True,
-    )
-    if res.stderr:
-        if "Must authenticate" in str(res.stderr):
-            raise AuthError(hostname=hostname, message=res.stderr)
-        else:
-            raise APIError(f"Error fetching repos for {org_name}: {res.stderr}")
+    entries_remaining = True
+    page_num = 1
+    while entries_remaining:
+        res = requests.get(
+            f"{BASE_URL}/orgs/{org_name}/repos",
+            params={
+                "page": page_num,
+                "per_page": 100,  # max value
+            },
+            headers=headers,
+        )
+
+        if res.status_code == 401 or res.status_code == 403:
+            raise AuthError(hostname, message=res.json())
+        if res.status_code != 200:
+            raise APIError(f"Error fetching repos for {org_name}: {res.json()}")
+
+        data = res.json()
+        if not isinstance(data, list):
+            raise JSON_SCHEMA_ERROR
+        entries_remaining = len(data) > 0
```

Comment:
nit: The length of the data doesn't really matter, but rather if there is data or not. Change this to `bool(data)` so we aren't going through the process of computing the length.

To be fair, this could be a premature optimization as I don't know if python already stores the length of strings since they are immutable.